### PR TITLE
Backport: Add 'isNewInstance' for updateAttributes

### PR DESCRIPTION
**Pull request by Amir-61**
* Add 'isNewInstance' to loaded hook for updateAttributes
Backport of #1094 

Connect to strongloop-internal/scrum-loopback#1030

@superkhau Could you please approve this PR so I can land it?

### CODE REVIEW

**Comment by Amir-61** [comment]
@superkhau 
>Please fix commit linter error on CI -- you need a blank line in between the list item and the Backport of line.

Done. PTAL. Thanks!

**Comment by superkhau** [comment]
@Amir-61 PR linter too while you're at it, says this PR needs rebasing.

**Comment by superkhau** [comment]
test please

**Comment by Amir-61** [comment]
@bajtos @superkhau 

Actually the simple backport PR of https://github.com/strongloop/loopback-datasource-juggler/pull/1094 was not working for juggler@2.x and was failing; for juggler@2.x I fixed (see: https://github.com/strongloop/loopback-datasource-juggler/pull/1105/commits/46d3439d6cda1e4187d3816cee7a5c47eefc39d7) load hook for `updateOrCreate` in this PR as well to make all tests pass with this 

So Please review this commit as well.

Thanks!

File: test/persistence-hooks.suite.js [hunk 1]
```diff
@@ -2014,6 +2014,7 @@ module.exports = function(dataSource, should, connectorCapabilities) {
             if (dataSource.connector.updateOrCreate) {
```
Comment (bajtos) [inline test/persistence-hooks.suite.js]: AFAICT, both branches report the same context data now and this `if` can be removed.

**Comment by bajtos** [comment]
> Actually the simple backport PR of #1094 was not working for juggler@2.x and was failing; for juggler@2.x I fixed (see: 46d3439) load hook for updateOrCreate in this PR as well to make all tests pass with this.

Do we need to forward-port 46d3439 to `master`?

**Comment by Amir-61** [comment]
@bajtos 
>Do we need to forward-port 46d3439 to master?

No I dont think so. `master` branch does already have `isNewInstance` for `loaded` hook for `updateOrCreate` which was implemented by commit: https://github.com/strongloop/loopback-datasource-juggler/commit/5ef444e45d72ff18df5c99818e2ab25b09748740 which seems @raymondfeng has not backported to `2.x`.

**Comment by Amir-61** [comment]
@slnode test please

**Comment by rmg** [comment]
@slnode test please

**Pull request merged**